File: JavaScript/M03_JavaScriptProgrammingBasics/L06_NestedLoops/Lab/P04_SumOfTwoNumbers.js
function sumOfTwoNumbers(input) {
    let index = 0;
    let n1 = parseInt(input[index++]);
    let n2 = parseInt(input[index++]);
    let magicNumber = parseInt(input[index++]);

    let count = 0, found = false;
    for (let x = n1; x <= n2; x++) {
        for (let y = n1; y <= n2; y++) {
            count++;
            if (x + y == magicNumber) {
                console.log(`Combination N:${count} (${x} + ${y} = ${x + y})`);
                found = true;
                break;
            }
        }
        if (found) { break; }
    }
    if (!found) {
        console.log(`${count} combinations - neither equals ${magicNumber}`);
    }
}

File: C++/M01_ProgrammingBasics/L06_NestedLoops/Lab/Solutions/P04_SumOfTwoNumbers.cpp
#include <iostream>
using namespace std;

int main() {
    int startNumber, endNumber, magicNumber;
    cin >> startNumber;
    cin >> endNumber;
    cin >> magicNumber;

    int count = 0;
    bool isFound = false;

    for (int x1 = startNumber; x1 <= endNumber; x1++) {
        for (int x2 = startNumber; x2 <= endNumber; x2++) {
            count++;
            int sum = x1 + x2;
            if (sum == magicNumber) {
                isFound = true;
                cout << "Combination N:" << count << " (" << x1 << " + " << x2 << " = " << magicNumber << ")";
                break;
            }
        }
        if (isFound) {
            break;
        }
    }
    if (!isFound) {
        cout << count << " combinations - neither equals " << magicNumber;
    }
    return 0;
}


// Second option

/*
#include <iostream>
using namespace std;

int main()
{
    int startNumber, endNumber, magicNumber;
    cin >> startNumber;
    cin >> endNumber;
    cin >> magicNumber;

    int count = 0;
    for (int x1 = startNumber; x1 <= endNumber; x1++) {
        for (int x2 = startNumber; x2 <= endNumber; x2++) {
            count++;
            int sum = x1 + x2;
            if (sum == magicNumber) {
                cout << "Combination N:" << count << " (" << x1 << " + " << x2 << " = " << magicNumber << ")";
                return 0;
            }
        }
    }
    cout << count << " combinations - neither equals " << magicNumber;

    return 0;
}
*/